src/java.base/share/classes/java/io/PushbackInputStream.java

```diff
@@ -405,4 +408,38 @@ private void implClose() throws IOException {
             buf = null;
         }
     }
+
+    @Override
+    public long transferTo(OutputStream out) throws IOException {
+        Objects.requireNonNull(out, "out");
+        ensureOpen();
+        if (lock != null) {
```

Comment:
PushbackInputStream's read methods are not synchronised so it doesn't make sense to introduce this for transferTo. In other words, I think you leave "clockLock" as it was.

Reply:
I have undone that change, so this PR only introduces the performance optimization now.